modules/utils/docker/bridge_settings.sh

```diff
@@ -50,20 +50,111 @@ if [ -f "$COMMS_PCB_VERSION_FILE" ]; then
 fi
 echo $eth_port
 
-# Set mtu back to 1500 to support e2e connectivity
-# TODO: Investigate this as it should still work with 1460
-ifconfig bat0 mtu 1500
-ifconfig br-lan mtu 1500
-
-brctl addif br-lan bat0 $eth_port
-echo $br_lan_ip
-ifconfig br-lan $br_lan_ip netmask "255.255.255.0"
-ifconfig br-lan up
-echo
-ifconfig br-lan
+create_dhcpd_config()
+{
+  SUBNET="$1"
+
+  cat > /etc/dhcp/dhcpd.conf <<- EOF
+  default-lease-time 600;
+  max-lease-time 7200;
+  ddns-update-style none;
+  authoritative;
+
+  subnet $SUBNET.0 netmask 255.255.255.0 {
+    range $SUBNET.100 $SUBNET.199;
+    option routers $SUBNET.1;
+  }
+EOF
+  cp /dev/null /var/lib/dhcp/dhcpd.leases
+}
+
+create_olsrd_config()
+{
+  wifidev="$1"
+  SUBNET="$2"
+  IPV6_PREFIX="$3"
+  cat > /etc/olsrd/olsrd.conf <<- EOF
+  LinkQualityFishEye   0
+
+  Interface "$wifidev"
+  {
+  }
+
+  IpVersion               4
+  LinkQualityFishEye      0
+  LinkQualityAlgorithm "etx_ffeth_nl80211"
+
+  LoadPlugin "/usr/lib/olsrd_arprefresh.so.0.1"  
+  {
+  }
+
+  Hna4
+  {
+    $SUBNET.0 255.255.255.0
+  }
+
+  Hna6
+  {
+    $IPV6_PREFIX:0 64
+  }
+EOF
+}
+
+create_radvd_config()
+{
+  IPV6_PREFIX="$1"
+  cat > /etc/radvd.conf <<- EOF
+  interface br-lan
+  {
+    AdvSendAdvert on;
+    MinRtrAdvInterval 3;
+    MaxRtrAdvInterval 10;
+    prefix $IPV6_PREFIX:0/64 {
+    };
+  };
+EOF
+}
+
+
+
+# Bridge ethernet and Mesh for Batman-avd and routing for QoS-OLSR
+if [ "$algo" = "olsr" ]; then
+  brctl addif br-lan "$eth_port"
+  IPV6_PREFIX=$( echo fd`dd if=/dev/urandom bs=7 count=1 status=none | xxd -p` | sed 's/\(....\)/\1:/g' )
+  ip addr add $SUBNET.1/24 dev br-lan && ip addr add $IPV6_PREFIX:1/64 dev br-lan
+  create_olsrd_config "wlp1s0" "$SUBNET" "$IPV6_PREFIX"
```

Comment:
We should find the interface name (i.e., wlxxxx) rather than hardcode it. Like in mcc_settings.sh. Same does for all other hardcoded wlp1s0 references in these changes.

Reply:
Hi! Names like wlp1s0 shouldn't be hardcoded indeed as they are likely to change when we swap the RPiCM4 for a SalukiPi for example since wlp1s0 means something like "WiFi card on PCI(e) port 1.0", but programmatically _finding_ the interface name might also be risky: maybe safer to have it as a configured variable name if that is not the case already?

Also I might have written that myself but upon careful consideration there should not be a `&&` preventing IPv6 configuration if IPv4 failed, especially since IPv4 is the more problematic one: I would recommend writing the two `ip` commands on separate lines so that if either fails then hopefully we at least still have the other :)

Reply:
We now seem to have $WIFI_INTERFACE but I have no idea where this var is coming from as it's not in the default mesh.conf. @francislin12 you will need to create another PR for tc_distro which creates the default mesh.conf